backend/src/main/java/gov/cdc/usds/simplereport/api/model/errors/TestEventSerializationFailureException.java

```diff
@@ -0,0 +1,32 @@
+package gov.cdc.usds.simplereport.api.model.errors;
+
+import graphql.ErrorClassification;
+import graphql.ErrorType;
+import graphql.GraphQLError;
+import graphql.language.SourceLocation;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Exception to throw when a TestEvent fails to serialize when saving to the ReportStream uploader
+ * queue
+ */
+public class TestEventSerializationFailureException extends RuntimeException
+    implements GraphQLError {
+
+  private static final long serialVersionUID = 1L;
+
+  public TestEventSerializationFailureException(UUID testEventId, String e) {
+    super(String.format("TestEvent failed to serialize with UUID %s: %s", testEventId, e));
+  }
+
+  @Override // should-be-defaulted unused interface method
+  public List<SourceLocation> getLocations() {
```

Comment:
for curiosity: what's the purpose of these methods? See that they're used in the class constructors for a couple of different error types but they don't seem to be invoked here. Wondering what they're here for / what they should be using (especially since Sonar seems to be complaining about returning null here.)

Reply:
I believe this is used by the graphQL framework, when it re-throws the exceptions. Specifically by the [GraphqlErrorHelper](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/GraphqlErrorHelper.java#L23)

My guess is that this is that returning `null` here might obfuscate the stack trace and make sure it doesn't come through to the frontend? But that's purely a guess based on the variable name.